pontoon/base/models.py

```diff
@@ -25,12 +25,101 @@
 log = logging.getLogger('pontoon')
 
 
+class UserTranslationsManager(models.Manager):
+    """
+    Provides various method to interact with larger sets of translations and their stats for user.
+    """
+
+    def _changed_translations_counts(self, start_date=None, end_date=None, filter=None, exclude=None):
+        """
+        Helper method, returns expression object which allows us to annotate querysets
+        with counts of translations.
+        Method supports exclusion of fields, you can exclude field if param contains
+        exclude__ prefix
+        e.g.
+        _changed_translations_counts(exclude__approved=True)
+        Will exclude "approved" field with True value.
+        """
+        translation_query = (
+            ~Q(translation__string=F('translation__entity__string'))
+            & ~Q(translation__string=F('translation__entity__string'))
+        )
+
+        if filter:
+            translation_query &= Q(**filter)
+
+        if exclude:
+            translation_query &= ~Q(**exclude)
+
+        if start_date and end_date:
+            translation_query &= Q(translation__date__range=(start_date, end_date))
+
+        # For each translation that matches the filter, return 1. Aggregate
+        # the sum of all those results to count the number of matching
+        # translations.
+        return Sum(
+                Case(
+                    When(translation_query, then=1),
+                    output_field=models.IntegerField(),
+                    default=0
+                    )
+                )
+
+    def get_changed_translations(self, start_date=None, end_date=None):
+        translations = (
+            Translation.objects
+            .exclude(user=None, string=F('entity__string'))
+            .exclude(string=F('entity__string_plural'))
+        )
+        if start_date and end_date:
+            translations = translations.filter(date__range=(start_date, end_date))
+        return translations
+
+    def get_approved(self, start_date=None, end_date=None):
+        return self.get_changed_translations(start_date, end_date).filter(approved=True)
+
+    def get_unapproved(self, start_date=None, end_date=None):
+        return self.get_changed_translations(start_date, end_date).exclude(approved=True, fuzzy=True)
+
+    def get_needs_work(self, start_date=None, end_date=None):
+        return self.get_changed_translations(start_date, end_date).filter(fuzzy=True)
+
+    def get_contributors_with_stats(self, start_date=None, end_date=None, limit=100):
```

Comment:
`with_translation_counts` is a more descriptive name; `get_contributors` is redundant given that this is called with `User.translations`, and we should be specific about what stats we're adding in case we want to get other stats at a later date.